pkgtools/texlive2pkg/files/texlive2pkg.8
.\"	$NetBSD: texlive2pkg.8,v 1.3 2019/08/27 06:21:18 markd Exp $
.\"
.\" Copyright (c) 2016
.\"	Mark Davies.  All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. Neither the name of the author nor the names of any contributors
.\"    may be used to endorse or promote products derived from this software
.\"    without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
.\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
.\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
.\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
.\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
.\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
.\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
.\" SUCH DAMAGE.
.\"
.\"
.Dd January 24, 2016
.Dt TEXLIVE2PKG 8
.Os
.Sh NAME
.Nm texlive2pkg
.Nd create a pkgsrc package for a texlive package
.Sh SYNOPSIS
.Nm
.Op Fl d
.Ar package-tarball
.Op Ar doc-tarball
.Sh DESCRIPTION
.Nm
takes a texlive
.Ar package-tarball
and optional
.Ar doc-tarball
that should be turned into
.Xr pkgsrc 7
packages under the current directory.
The generated packages include a
.Pa Makefile
and a mostly complete set of files to get the package going.
However, the files should be reviewed and perhaps adjusted manually after
.Nm
has finished its job.
.Pp
.Nm
is intended to help the process of creating a package but is not
intended to fly on autopilot.
.Sh OPTIONS
The following command-line options are supported:
.Bl -tag -width indent
.It Fl d
Derive the doc-tarball path from the package-tarball 
by replacing the .tar.xz with .doc.tar.xz.
.Sh EXAMPLES
Use the following commands to create a
.Xr pkgsrc 7
package for the texlive package
.Ar foo :
.Dl cd pkgsrc/print
.Dl texlive2pkg -d foo.tar.xz
Remember to test the resulting package.
.Sh SEE ALSO
.Xr pkgsrc 7
.Sh AUTHORS
.Nm
was written by
.An Mark Davies Aq Mt markd@NetBSD.org .
.Sh BUGS
.Nm
does not create a finished package; instead, it facilitates the task.
However, the resulting files must be reviewed and possibly edited by
hand.
Although dependencies are included in the generated
.Pa Makefile ,
they are only comments; no attempt is made to find corresponding
packages within
.Xr pkgsrc 7 .
As a consequence, the category of dependencies is also not determined.


doc/NEWS
# $NetBSD: NEWS,v 1.12 2019/04/24 13:19:31 maya Exp $

PKGSRC NEWS
===========

  This file contains information about important infrastructural changes in
  pkgsrc. The intended readers are the pkgsrc developers as well as anyone
  tracking pkgsrc-current.

python 3.7 default [maya 2019-04-24]
	As Python 2.x versions are nearing end of life, we have switched
	the default to a newer version.
	Python 2.7 packages are still available, and most python packages
	are built for all available python versions.

Lua multiversion support [alnsn 2014-05-03]
	Lua packages lang/lua51 and lang/lua52 have been modified
	to coexist without conflicts, infrastructure for supporting
	multiple versions of Lua has been put in place and all
	dependent packages have been adapted to the new infrastructure.

SMF support is introduced [jperkin 2014-03-11]

Unprivileged staged installation is default [asau 2012-08-14]
	pkgsrc has changed to use unprivileged staged installation by default.
	This makes it possible to prevent some installation mistakes.
	It also makes detection of package conflicts easier.
	Previously you had to set PKG_DESTDIR_SUPPORT to "user-destdir"
	to invoke this functionality. Set PKG_DESTDIR_SUPPORT to "none"
	to revert to direct installation.

buildlink3.mk layout changed [joerg 2009-03-20]
	The buildlink3.mk files have been reorganized to reduce
	redundant parsing for deeply nested buildlink3 trees.
	The createbuildlink package has been updated to create
	the new format.

PLIST_VARS introduced [jlam 20080413]
	PLIST_VARS is a simpler way to do conditional PLIST entries.
	The old method of doing e.g. PLIST_SUBST+= FOO="@comment " can
	now be done by unconditionally setting PLIST_VARS+=foo and
	conditionally setting PLIST.foo=yes. This will result in substitution
	of ${PLIST.foo} in the PLIST with either "" or "@comment ".

SKIP_PORTABILITY_CHECK is obsolete [rillig 20080220]
	After being deprecated for about 16 months, this early misdesign
	has finally been obsoleted and removed from pkgsrc.

Check for stripped/unstripped binaries [rillig 20080213]
	A new check has been added in mk/check/check-stripped.mk to
	assure that the installed binaries conform to the setting of the
	INSTALL_UNSTRIPPED variable. It is not enabled by default due to
	lack of testing.

OpenSSL update [tnn 20080118]
	OpenSSL in pkgsrc switched from the 0.9.7 to the 0.9.8 branch. To
	accomodate for this, 596 packages dependent on openssl had their
	revisions bumped.

pkgsrc-2007Q4 released [agc 20080105]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2008/01/05/0000.html

Simplifying meta-packages [rillig 20071219]
	The variable META_PACKAGE was introduced. This simplifies creation
	and maintenance of packages that do not install files.

Replace a package with different name [gdt 20071107]
	The variable OLDNAME was introduced for "make replace" to
	replace a package with a name other than the new one, via
	"make replace OLDNAME=oldpkgname".  Read the announcement at:
	http://mail-index.netbsd.org/tech-pkg/2007/11/07/0002.html
	
pkgsrc-2007Q3 released [agc 20071015]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/10/15/0000.html

Features framework [jlam 20070907]
	A "system features" framework has been added. It will be used to
	automatically supply missing "basic" headers and libraries for old
	systems. This is supported through the USE_FEATURES variable.

X.org 6.9 removed [joerg 20070817]
	monolithic X.org 6.9 was removed from pkgsrc. It was obsoleted by the
	modular X.org (7.x) packages. Transitioning users should remove all old
	X.org packages and set X11_TYPE=modular in mk.conf.

Emulator framework [jlam 20070729]
	A new emulator framework has been added. This will handle all
	binary-only packages that require emulation on the native OS.
	Please see pkgsrc/mk/emulator/README.

audit-packages merged with pkg_install [adrianp 20070714]
	The utility for auditing installed packages for security problems,
	audit-packages, has been rewritten in C. It is now part of the
	pkgtools/pkg_install package.
	
pkgsrc-2007Q2 released [agc 20070701]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/07/01/0000.html

New framework for bulk builds [joerg 20070619]
	pbulk, the new automated pkgsrc bulk build framework was imported.

pkgsrc-2007Q1 released [agc 20070419]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/04/19/0000.html

pkgsrc-2006Q4 released [agc 20070104]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2007/01/04/0000.html

modular X.org imported [joerg 20061114]
	The modular X.org packages were imported.

pkgsrc-2006Q3 released [agc 20061001]
	A new stable branch was released. Read the announcement at:
	http://mail-index.netbsd.org/netbsd-announce/2006/10/01/0000.html


pkgtools/texlive2pkg/files/texlive2pkg.sh
#!/bin/sh
# texlive2pkg
# $NetBSD: texlive2pkg.sh,v 1.2 2019/08/27 06:21:18 markd Exp $
#
# Copyright (c) 2016
#       Mark Davies.  All rights reserved.
#
# Redistribution and use in source and binary forms, with or without
# modification, are permitted provided that the following conditions
# are met:
# 1. Redistributions of source code must retain the above copyright
#    notice, this list of conditions and the following disclaimer.
# 2. Redistributions in binary form must reproduce the above copyright
#    notice, this list of conditions and the following disclaimer in the
#    documentation and/or other materials provided with the distribution.
# 3. Neither the name of the author nor the names of any contributors
#    may be used to endorse or promote products derived from this software
#    without specific prior written permission.
#
# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
# ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
# SUCH DAMAGE.
#
#
# Create a texlive package in the current directory

NAME=texlive2pkg
TDIR=/tmp/tl.$$
DISTDIR=@DISTDIR@
LIBEXECDIR=@PREFIX@/libexec/texlive2pkg
MAKE=@MAKE@
FTP=@FTP@
pkgtarball=""
doctarball=""
dodoc=0
pkgurl=0
docurl=0

USAGE="${NAME} [-d] package-tarball [doc-tarball] -- create a texlive package for pkgsrc"

if [ "$1" = "-d" ]; then
  dodoc=1
  shift
fi

if [ ${#} -eq 1 ]; then
  pkgtarball=$1
  if [ $dodoc -eq 1 ]; then
    doctarball=${pkgtarball%%.tar.xz}.doc.tar.xz
  fi
elif [ ${#} -eq 2 ]; then
  pkgtarball=$1
  doctarball=$2
else
  echo ${USAGE}
  exit 1
fi

case $pkgtarball in
file:/*.tar.xz | ftp:/*.tar.xz | http:/*.tar.xz | https:/*.tar.xz )
   pkgurl=1
   break ;;
/*.tar.xz)
   break ;;
*.tar.xz)
   pkgtarball=$PWD/$pkgtarball
   break ;;
*)
   echo ${USAGE}
   exit 1
esac

case "$doctarball" in
"")
   break ;;
file:/*.doc.tar.xz | ftp:/*.tar.xz | http:/*.doc.tar.xz | https:/*.doc.tar.xz )
   docurl=1
   break ;;
/*.doc.tar.xz)
   break ;;  
*.doc.tar.xz)
   doctarball=$PWD/$doctarball
   break ;;
*)
   echo ${USAGE}
   exit 1
esac

mkdir $TDIR

if [ $pkgurl -eq 1 ]; then
  (cd $TDIR; $FTP $pkgtarball )
  pkgtarball=$TDIR/`basename $pkgtarball`
fi
if [ $docurl -eq 1 ]; then
  (cd $TDIR; $FTP $doctarball )
  doctarball=$TDIR/`basename $doctarball`
fi

if [ ! -f $pkgtarball ]; then
  echo "File not found - $pkgtarball"
  exit 1
fi

(cd $TDIR ; xzcat $pkgtarball | tar xf - tlpkg)
eval `$LIBEXECDIR/texlive.pkg $TDIR/tlpkg/tlpobj/*.tlpobj`
mkdir -p $DISTDIR/tex-$PKG-$REV
cp $pkgtarball $doctarball $DISTDIR/tex-$PKG-$REV/
(cd tex-$PKG; $MAKE makesum)
if [ -f "$doctarball" ]; then
   (cd $TDIR; xzcat $doctarball | tar xf - tlpkg)
   $LIBEXECDIR/texlive.pkg.doc $PKG $REV "$VERSION" "$LICENSE" "$HOMEPAGE" < $TDIR/tlpkg/tlpobj/${PKG}.doc.tlpobj && \
   (cd tex-$PKG-doc; $MAKE makesum)
fi
rm -rf $TDIR
